ztools/ztls/common.go

```diff
@@ -949,3 +982,157 @@ type ClientHelloInfo struct {
 	// connection to fail.
 	Conn net.Conn
 }
+
+func (info *ClientHelloInfo) MarshalJSON() ([]byte, error) {
+	aux := struct {
```

Comment:
so it might make sense to split out this JSON to match the shape of an actual client hello. I realize that Golang is dumb and makes everything flat, but it'd be nice to at least see extensions in something called `extensions`.